### ISSUE TITLE
Unable to run wireless on Mac

### ISSUE THREAD

**Issue by ddoak** [davidjsherman/inirobot-scratch-thymioII#10]
I am unable to get 0.73 (or earlier) release to work with a wireless thymio and Mac (OSX 10.9.5).
I've never had a problem on Mac when USB tethered.  I can also get it to work both wireless and tethered with the 0.73 release on a Win10 machine.

I have installed the latest aseba 1.5.3 / scratch thymio bundle - 

Here's what I'm trying - Is this the correct sequence?

- Insert wireless dongle
- Power on thymio
- Run asebastudio (and connect to serial wireless thymio) - thymio responds (wireless led flashes - and at this point could be used with asebastudio/VPL/blockly)
- Run Scratch2-ThymioII helper
- Run Scratch
- Load Thymio-II.sb2 (or extensions)

At this point the scratch2-thymioII extension blocks are shown as active (green button) but the thymio does not respond to the scratch command blocks

I have tried this on two machines (iMac and MacBook both - OSX 10.9.5).
Thymio firmware is current (v10).

**Comment by davidjsherman** [comment]
Are you using Scratch 447 also? The Scratch protocol changed around version 446. 

Cheers,

David James Sherman
Directeur de recherche

Inria Bordeaux Sud-Ouest
351 cours de la Libération
33405 Talence Cedex – France 
Tel : +33 (0)5 24 57 40 00
Fax : +33 (0)5 24 57 41 24

www.inria.fr

**Comment by ddoak** [comment]
@davidjsherman 
Yes - using Scratch v447

**Comment by davidjsherman** [comment]
Could you please try the examples "ThymioII sensors" and "Star with Dog"?

Is the red LED in the dongle on, flickering, or off? If it is on all the time, then it's communication buffer is full and you need to reinsert it to reset it.

The red LED inside the Thymio on the RF card should be flickering at 10 Hz, as the robot sends its state to the helper.

The green spot in Scratch shows that the helper is active, but doesn't guarantee that it is talking to the robot.

Thanks,
David

**Comment by ddoak** [comment]
Tried the "ThymioII sensors" file. No luck.
Dongle LED and Thymio RF LED both pulsing. 
[https://www.dropbox.com/s/ds1ijcgds2msx2y/s2t%20test%2001%20480.mov?dl=0](url)

**Comment by ddoak** [comment]
Quick thought - my wireless thymios don't use the default ids (to stop confusion when children are using them at school) - might that be an issue?

**Comment by ddoak** [comment]
Just tried resetting the wireless dongle configuration to defaults (channel 1, network identifier 0x404F).
No difference.

**Comment by davidjsherman** [comment]
> Quick thought - my wireless thymios don't use the default ids (to stop confusion when children are using them at school) - might that be an issue?

No, the helpers asebahttp and asebascratch will use any ids and will even renumber them if there are conflicts.

If you have the command-line tool “curl” installed, could you try:

  curl http://localhost:3000/nodes

and

  curl http://localhost:3000/nodes/thymio-II

The first should show a list of robot names and identifiers, the second should show everything that it knows about the first robot named “thymio-II”, in JSON format. That will let us know whether the helper sees any robots at all.

**Comment by davidjsherman** [comment]
I can't tell in you video whether you stopped Aseba Studio or not — if asebascratch can't get exclusive access it will fail silently. You can see this if you run it from the command line, e.g.
```
% /Applications/Aseba/Scratch/Scratch2-ThymioII.app/Contents/MacOS/asebascratch --aesl /Applications/Aseba/Scratch/Scratch2-ThymioII.app/Contents/Resources/thymio_motion.aesl --http 3000 'ser:name=Thymio'
Found Thymio on port /dev/cu.usbmodem621
HttpInterface can't connect target ser:name=Thymio: Cannot lock serial port.

**Comment by ddoak** [comment]
I didn't stop it - am I supposed to quit it before running the helper?

**Comment by davidjsherman** [comment]
Testing with `curl` should give:
```
% curl -D - http://localhost:3000/nodes
HTTP/1.1 200 OK
Content-Length: 70
Content-Type: application/json
Access-Control-Allow-Origin: *

[{"node":50386,"name":"thymio-II","protocolVersion":5,"aeslId":50386}]

% curl -D - http://localhost:3000/nodes/thymio-II | fold
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2534  100  2534    0     0   155k      0 --:--:-- --:--:-- --:--:--  164k
HTTP/1.1 200 OK
Content-Length: 2534
Content-Type: application/json
Access-Control-Allow-Origin: *

{"node":50386,"name":"thymio-II","protocolVersion":5,"aeslId":50386,"bytecodeSiz
e":1534,"variablesSize":620,"stackSize":32,"namedVariables":{"Qid":4,"Qnx":1,"Qp
c":1,"QspL":4,"QspR":4,"Qtime":4,"R_state":27,"R_state.do":1,"_fwversion":2,"_id
":1,"_imot":2,"_productId":1,"_vbat":2,"acc":3,"acc._tap":1,"angle.back":1,"angl
e.front":1,"angle.ground":1,"button.backward":1,"button.center":1,"button.forwar
d":1,"button.left":1,"button.right":1,"buttons._mean":5,"buttons._noise":5,"butt
ons._raw":5,"distance.back":1,"distance.front":1,"event.args":32,"event.source":
1,"mic._mean":1,"mic.intensity":1,"mic.threshold":1,"motor.left.pwm":1,"motor.le
ft.speed":1,"motor.left.target":1,"motor.right.pwm":1,"motor.right.speed":1,"mot
or.right.target":1,"odo.degree":1,"odo.delta":1,"odo.theta":1,"odo.x":1,"odo.y":
1,"prox.comm.rx":1,"prox.comm.rx._intensities":7,"prox.comm.rx._payloads":7,"pro
x.comm.tx":1,"prox.ground.ambiant":2,"prox.ground.delta":2,"prox.ground.reflecte
d":2,"prox.horizontal":7,"rc5.address":1,"rc5.command":1,"temperature":1,"timer.
period":2,"tmp":9},"localEvents":{"button.backward":"Backward button status chan
ged","button.left":"Left button status changed","button.center":"Center button s
tatus changed","button.forward":"Forward button status changed","button.right":"
Right button status changed","buttons":"Buttons values updated","prox":"Proximit
y values updated","prox.comm":"Data received on the proximity communication","ta
p":"A tap is detected","acc":"Accelerometer values updated","mic":"Fired when mi
crophone intensity is above threshold","sound.finished":"Fired when the playback
 of a user initiated sound is finished","temperature":"Temperature value updated
","rc5":"RC5 message received","motor":"Motor timer","timer0":"Timer 0","timer1"
:"Timer 1"},"constants":{"QUEUE":4},"events":{"Q_add_motion":4,"Q_cancel_motion"
:1,"Q_motion_added":5,"Q_motion_cancelled":5,"Q_motion_started":5,"Q_motion_ende
d":5,"Q_motion_noneleft":1,"Q_set_odometer":3,"V_leds_prox_h":8,"V_leds_circle":
8,"V_leds_top":3,"V_leds_bottom":4,"V_leds_prox_v":2,"V_leds_buttons":4,"V_leds_
rc":1,"V_leds_temperature":2,"V_leds_sound":1,"A_sound_freq":2,"A_sound_play":1,
"A_sound_system":1,"A_sound_replay":1,"A_sound_record":1,"M_motor_left":1,"M_mot
or_right":1,"R_state_update":27,"Q_reset":0,"scratch_arc":3,"scratch_change_leds
":2,"scratch_change_speed":2,"scratch_clear_leds":0,"scratch_move":2,"scratch_ne
xt_dial":0,"scratch_next_dial_limit":1,"scratch_set_dial":1,"scratch_set_leds":2
,"scratch_start":2,"scratch_stop":0,"scratch_turn":2}}

**Comment by ddoak** [comment]
Davids-MacBook-Pro:~ ddoak$ curl -D - http://localhost:3000/nodes
HTTP/1.1 404 Not Found
Content-Length: 2
Content-Type: application/json
Access-Control-Allow-Origin: *

[]Davids-MacBook-Pro:~ ddoak$

**Comment by davidjsherman** [comment]
Yes, you need to give exclusive access to the helper.

If you absolutely need to connect the Studio at the same time, you can use asebascratch as a switch:
```
/Applications/Aseba/Scratch/Scratch2-ThymioII.app/Contents/MacOS/asebascratch --aesl /Applications/Aseba/Scratch/Scratch2-ThymioII.app/Contents/Resources/thymio_motion.aesl --http 3000 --port 33333 'ser:name=Thymio'
```
and connect Aseba Studio to TCP target tcp:;port=33333. But this is dangerous because unless you load exactly the same `thymio_motion.aesl` program with Aseba Studio, things will start to fail horribly!

**Comment by davidjsherman** [comment]
Yes, `[]` is the empty list: asebascratch has found no robots.

**Comment by ddoak** [comment]
So in my original post on this issue:


    Insert wireless dongle
    Power on thymio
    Run asebastudio (and connect to serial wireless thymio) - thymio responds (wireless led flashes - and at this point could be used with asebastudio/VPL/blockly)
    Run Scratch2-ThymioII helper
    Run Scratch
    Load Thymio-II.sb2 (or extensions)

After 'Run asebastudio'

I should quit it again? Once the dongle and wireless thymio LEDs are pulsing?

**Comment by davidjsherman** [comment]
You don't need to run Aseba Studio at all. If you do run it, you need to quit it before starting the helper.

**Comment by ddoak** [comment]
Doh!

**Comment by ddoak** [comment]
In my defence, I thought it used to work like that in the earlier versions...

**Comment by ddoak** [comment]
Just tried and it works. Can we pretend this issue never happened?

**Comment by davidjsherman** [comment]
What issue?
Glad it works! Let me know if the examples do what you expect. I would love to add your gamepad example in wireless mode. Now that the accelerometer tilt is sent to Scratch, you can make the turret rotate

**Comment by ddoak** [comment]
Absolutely - there is no issue.

However - I definitely owe you some testing and good examples!

Funny - because of my experiencse when I was randomly muddling around trying to get v0.6 running (a year ago) - I got into the habit of always running asebastudio before starting the scratch2thymioII helper - which of course doesn't cause any problems with the tethered USB cable. I just kept on doing the same thing when trying the wireless thymio - and then assuming that it was just not stable yet... when in fact I was stopping it from working.
